jax/scipy/optimize/bfgs_minimize.py

```diff
@@ -0,0 +1,140 @@
+  nhev: int  # total number of hessian evaluations
+  x_k: jnp.ndarray  # A tensor containing the last argument value found during the search. If the search converged,
+  # then this value is the argmin of the objective function.
+  f_k: jnp.ndarray  # A tensor containing the value of the objective
+  # function at the `position`. If the search
+  # converged, then this is the (local) minimum of
+  # the objective function.
+  g_k: jnp.ndarray  # A tensor containing the gradient of the objective function at the `final_position`.
+  # If the search converged the l2-norm of this tensor should be below the tolerance.
+  H_k: jnp.ndarray  # A tensor containing the inverse of the estimated Hessian.
+
+
+def bfgs_minimize(func, x0, options=None, _nojit=False):
+  """
+  The BFGS algorithm from
+      Algorithm 6.1 from Wright and Nocedal, 'Numerical Optimization', 1999, pg. 136-143
+
+      Notes:
+          We utilise boolean arithmetic to avoid jax.cond calls which don't work on accelerators.
+          A side effect is that we perform more gradient evaluations than scipy's BFGS
+      func: callable
+          Function of the form f(x) where x is a flat ndarray and returns a real scalar. The function should be
+          composed of operations with vjp defined. If func is jittable then bfgs_minimize is jittable. If func is
+          not jittable, then _nojit should be set to True.
+
+      x0: ndarray
+          initial variable
+      options: Optional dict of parameters
+          maxiter: int
+              Maximum number of evaluations
+          g_tol: flat
+              Terminates minimization when |grad|_2 < g_tol
+          ls_maxiter: int
+              Maximum number of linesearch iterations
+      _nojit: bool
+          Whether to use pythonic control flow so that func without XLA ops can be used. It is also very useful to
+          set _nojit=True in order to perform debugging.
+
+  Returns: BFGSResults
+
+  """
+  if options is None:
+    options = dict()
+  maxiter: Optional[int] = options.get('maxiter', None)
+  analytic_initial_hessian: bool = options.get('analytic_initial_hessian', False)
+  g_tol: float = options.get('g_tol', 1e-5)
+  ls_maxiter: int = options.get('ls_maxiter', 10)
+
+  state = BFGSResults(converged=False,
+                      failed=False,
+                      k=0,
+                      nfev=0,
+                      ngev=0,
+                      nhev=0,
+                      x_k=x0,
+                      f_k=None,
+                      g_k=None,
+                      H_k=None)
+
+  if maxiter is None:
+    maxiter = jnp.inf
+
+  D = x0.shape[0]
+
+  if analytic_initial_hessian:
+    hess = jax.hessian(func, argnums=0)
+    initial_B = hess(x0)
+    initial_H = jnp.linalg.pinv(initial_B)
+    # TODO: experimental, should remove
+    if jnp.any(jnp.linalg.eigvals(initial_B) <= 0):
+      initial_H = jnp.eye(D)
+    state = state._replace(nhev=state.nhev + 1)
```

Comment:
We want to remove this whole block, right?

Reply:
Yep, done.

Reply:
It looks like you may have forgotten to push your latest changes to github?

Reply:
Yes, now it's done.